internal/controllers/device_config_controller.go

```diff
@@ -294,34 +252,29 @@ func (d *DeviceConfigController) GetConfigURLs(c *fiber.Ctx) error {
 	vin := c.Params("vin")
 
 	// Set up gRPC call to retrieve UserDevice by VIN
-	req := &pb.GetUserDeviceByVINRequest{Vin: vin}
-	ud, err := d.usersClient.GetUserDeviceByVIN(context.Background(), req)
+	ud, err := d.userDeviceSvc.GetUserDeviceByVIN(context.Background(), vin)
 	if err != nil {
 		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
 			"error": fmt.Sprintf("Failed to retrieve user device for VIN: %s", vin),
 		})
 	}
+	// Query templates, filter by protocol and powertrain
```

Comment:
There could be cases where powertrain and/or canprotocol could be empty/nil - in those cases let's set them to a default: ICE and CAN11_500